instrumentation/rack/lib/opentelemetry/instrumentation/rack/middlewares/event_handler.rb

```diff
@@ -109,7 +109,7 @@ def on_finish(request, response)
             span = OpenTelemetry::Instrumentation::Rack.current_span
             return unless span.recording?
 
-            add_response_attributes(span, response) if response
+            add_response_attributes(span, response) if response && !request.env.key?('rack.hijack?')
```

Comment:
@j15e The more I think about your comments in slack, the more I agree with this implementation:

> I see, so maybe changing the status check to > 500 would be a simpler and more pragmatic fix, is that what you think?[^1]

When we take a closer look at the spec:

> [Span Status](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.37.0/specification/trace/api.md#set-status) MUST be left unset if HTTP status code was in the 1xx, 2xx or 3xx ranges, unless there was another error (e.g., network error receiving the response body; or 3xx codes with max redirects exceeded), in which case status MUST be set to Error.
> ...
> For HTTP status codes in the 5xx range, as well as any other code the client failed to interpret, span status SHOULD be set to Error.

I interpret that to mean that the code should only set the status to error in cases where the response is in the `5xx` ranges . There are other non-5xx cases that the code isn't currently accounting for but the spec isn't terribly explicit about it other than the case of the Max Redirects Exceeded, which is usually a client side constraint.  

Given the description the spec I think it is safe to change the code to look for Bad HTTP status codes in the 5xx range and leaving the span status unset in all other cases. 

This will remove our explicit checks for the Rack Hijack implementation and allow the response context propagators to execute. 

[^1]: https://cloud-native.slack.com/archives/C01NWKKMKMY/p1729384339814629?thread_ts=1729277160.681719&cid=C01NWKKMKMY
[^2]: https://opentelemetry.io/docs/specs/semconv/http/http-spans/#status